modules/ardrivo/test/Arduino.cxx

```diff
@@ -0,0 +1,322 @@
+    REQUIRE(isAlphaNumeric('a'));
+    REQUIRE(isAlphaNumeric('1'));
+    REQUIRE(!isAlphaNumeric('!'));
+}
+
+//shit does not work
+TEST_CASE("Is the character a ascii", "[isAscii]") {
+    /*
+    char num = 'a';
+    REQUIRE(isAscii(num)); 
+    */
+}
+
+TEST_CASE("Check if the character a control character", "[isControl") {
+    REQUIRE(isControl('\t'));
+    REQUIRE(isControl('\n'));
+    REQUIRE(isControl('\v'));
+}
+TEST_CASE("Checks if the charahcter a digit" "[isDigit]") {
+    REQUIRE(isDigit('1'));
+    REQUIRE(!isDigit('a'));
+    REQUIRE(!isDigit('#'));
+}
+
+TEST_CASE("Checks if a character is a digit" "[isDigit]") {
+    REQUIRE(isDigit('1'));
+    REQUIRE(!isDigit('a'));
+    REQUIRE(!isDigit('#'));
+}
+
+TEST_CASE("Checks if a character is a graphical represention" "[isGraph]") {
+    REQUIRE(isGraph('a'));
+    REQUIRE(isGraph('1'));
+    REQUIRE(!isGraph(' '));
+}
+
+TEST_CASE("Checks if a character is hexadecimal", "[isHexadecimalDigit]") {
+    REQUIRE(isHexadecimalDigit('a'));
+    REQUIRE(!isHexadecimalDigit('!'));
+}
+
+TEST_CASE("Checks if a character is lowercase", "[isLowerCase") {
+    REQUIRE(isLowerCase('a'));
+    REQUIRE(!isLowerCase('A'));
+}
+
+TEST_CASE("Checks if a character is printable", "[isPrintable]") {
+    REQUIRE(isPrintable('a'));
+    REQUIRE(isPrintable('1'));
+    REQUIRE(isPrintable(' '));
+}
+
+TEST_CASE("Checks if a character is a punctuation" "[isPunct") {
+    REQUIRE(!isPunct('a'));
+    REQUIRE(isPunct(','));
+    REQUIRE(isPunct('?'));
+}
+
+TEST_CASE("Checks if a character contains white-space", "[isSpace]") {
+    REQUIRE(!isSpace('a'));
+    REQUIRE(isSpace('\t'));
+    REQUIRE(isSpace(' '));
+}
+
+TEST_CASE("Checks if a character is uppercase" "[isUpperCase") 
+{
+    REQUIRE(isUpperCase('A'));
+    REQUIRE(!isUpperCase('a'));
+}
+
+TEST_CASE("Checks if a character is a blank character" "[isWhiteSpace]") 
+{
+    REQUIRE(isWhitespace(' '));
+    REQUIRE(isWhitespace('\t'));
+    REQUIRE(!isWhitespace('a'));
+}
+
+TEST_CASE("Checks if it returns a random number" "[isRandom]") 
+{
+    auto i = GENERATE(take(100, filter([](int i) { return i; }, random(-100, 100))));
+    REQUIRE(i < 100);
+    REQUIRE(i > -100);
+}
+
+//The test is lying, since it is testing a C random function rather than the Arduino ones
+
+TEST_CASE("Test randomSeed", "[randomSeed]") 
+{
+    long arr[100];
+    long arr2[100];
+    long arr3[100];
+    auto count = 0;
+    randomSeed(2);
+
+    for (int i = 0; i < 100; i++)
+    {
+        arr[i] = random(100);
+    }
+    randomSeed(3);
+
+    for (int i = 0; i < 100; i++)
+    {
+        arr2[i] = random(100);
+    }
+
+    for (int i = 0; i < 100; i++)
+    {
+        if (arr[i] == arr2[i])
+        {
+            count++;
+        }
+    }
+    REQUIRE(count < 3);
+
+    randomSeed(2);
+
+    for (int i = 0; i < 100; i++)
+    {
+        arr3[i] = random(100);
+    }
+
+    for (int i = 0; i < 100; i++)
+    {
+        REQUIRE(arr[i] == arr3[i]);
+    }
```

Comment:
pretty sure you can just `arr == arr3` (on std::array and std::vector) otherwise do `ranges::equal` (<range/v3/algorithm/equal.hpp> iirc) or [`std::equal`](https://en.cppreference.com/w/cpp/algorithm/equal)

Reply:
+1 Indeed `std::array` has lexicographical comparison operators